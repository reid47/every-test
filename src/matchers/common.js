export const formatValue = received => {
  if (received == null) return String(received);
  if (received instanceof HTMLElement) return received.outerHTML;

  try {
    return JSON.stringify(received);
  } catch {
    return received.toString();
  }
};
