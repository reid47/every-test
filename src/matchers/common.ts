export const formatValue = (received: any): string => {
  if (received == null) return String(received);
  if (received instanceof HTMLElement) return received.outerHTML;

  try {
    return JSON.stringify(received);
  } catch {
    return received.toString();
  }
};
