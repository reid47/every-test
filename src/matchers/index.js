export { default as toBeChecked } from './toBeChecked';
export { default as toBeDisabled } from './toBeDisabled';
export { default as toBeFocused } from './toBeFocused';
export { default as toBeHidden } from './toBeHidden';
export { default as toContainText } from './toContainText';
export { default as toExist } from './toExist';
export { default as toHaveAttr } from './toHaveAttr';
export { default as toHaveClass } from './toHaveClass';
export { default as toHaveProp } from './toHaveProp';
export { default as toHaveText } from './toHaveText';
export { default as toHaveValue } from './toHaveValue';
