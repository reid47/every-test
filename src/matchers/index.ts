export { default as toBeChecked } from './toBeChecked';
export { default as toBeDisabled } from './toBeDisabled';
export { default as toBeEmpty } from './toBeEmpty';
export { default as toBeFocused } from './toBeFocused';
export { default as toBeHidden } from './toBeHidden';
export { default as toContainText } from './toContainText';
export { default as toExist } from './toExist';
export { default as toHaveAttribute } from './toHaveAttribute';
export { default as toHaveClass } from './toHaveClass';
export { default as toHaveTagName } from './toHaveTagName';
export { default as toHaveText } from './toHaveText';
export { default as toHaveValue } from './toHaveValue';
