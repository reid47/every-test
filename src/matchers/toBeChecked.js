export default function toBeChecked(received) {
  const pass = !!received.checked;
  const message = pass ? `Expected element not to be checked.` : `Expected element to be checked.`;
  return { pass, message };
}
