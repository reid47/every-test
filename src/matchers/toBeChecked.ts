export default function toBeChecked(received: HTMLElement) {
  const pass = received instanceof HTMLInputElement && received.checked;
  const message = pass ? `Expected element not to be checked.` : `Expected element to be checked.`;
  return { pass, message };
}
