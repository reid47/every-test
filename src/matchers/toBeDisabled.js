export default function toBeDisabled(received) {
  const pass = !!received.disabled;

  const message = pass
    ? 'Expected element not to be disabled, but it was.'
    : 'Expected element to be disabled, but it was not.';

  return { pass, message };
}
