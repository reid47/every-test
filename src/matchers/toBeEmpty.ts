export function toBeEmpty(received) {
  const pass = received.innerHTML === '';

  const message = pass
    ? 'Expected element not to be empty, but it was.'
    : `Expected element to be empty, but it has content: ${received.innerHTML}`;

  return { pass, message };
}
