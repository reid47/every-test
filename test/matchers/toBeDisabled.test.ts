import { toBeDisabled } from '../../src/matchers/toBeDisabled';

describe('toBeDisabled', () => {
  test('passing', () => {
    const input = document.createElement('input');
    input.disabled = true;

    const result = toBeDisabled(input);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing', () => {
    const input = document.createElement('input');
    input.disabled = false;

    const result = toBeDisabled(input);
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });
});
