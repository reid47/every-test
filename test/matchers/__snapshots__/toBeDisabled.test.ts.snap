// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeDisabled failing 1`] = `"Expected element to be disabled, but it was not."`;

exports[`toBeDisabled passing 1`] = `"Expected element not to be disabled, but it was."`;
